src/index.ts

```diff
@@ -50,25 +65,34 @@ function routePlugin(userOptions: UserOptions = {}): Plugin {
     configResolved(_config) {
       config = _config
       options.root = config.root
-      options.pagesDirPath = normalizePath(resolve(config.root, options.pagesDir))
-      debug('pagesDirPath', options.pagesDirPath)
     },
     resolveId(id) {
       if (id === ID)
         return ID
     },
     async load(id) {
       if (id === ID) {
-        debug('Loading files...')
-
-        filesPath = await getPagesPath(options)
-
-        debug('FilesPath: %O', filesPath)
-
-        if (!generatedRoutes)
-          generatedRoutes = generateRoutes(filesPath, options)
-
-        debug('Routes: %O', generatedRoutes)
+        debug('Loading...')
+        debug('Loading Gen Routes: %O', generatedRoutes)
+
+        if (!generatedRoutes) {
+          generatedRoutes = []
+          filesPath = []
+          options.pagesDirPaths = []
```

Comment:
I think you can move this one to plugin root function, don't need to be a resolved options.